src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            first = new Node<>(null, value, null);
+            last = first;
+        } else {
+            Node<T> newNode = new Node<>(last, value, null);
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, first);
+            first.prev = newNode;
+            first = newNode;
+        } else {
+            checkIndex(index);
+            Node<T> targetNode = getNode(index);
+            Node<T> node = new Node<>(targetNode.prev, value, targetNode);
+            targetNode.prev.next = node;
+            targetNode.prev = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> targetNode = getNode(index);
+        T returnElement = targetNode.value;
+        targetNode.value = value;
+        return returnElement;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> targetNode = getNode(index);
+        return unlink(targetNode);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = first;
+        if (object == null) {
+            while (node != null) {
+                if (node.value == null) {
+                    unlink(node);
+                    return true;
+                }
+                node = node.next;
+            }
+        } else {
+            while (node != null) {
+                if (node.value.equals(object)) {
+                    unlink(node);
+                    return true;
+                }
+                node = node.next;
+            }
+        }
```

Comment:
You can avoid if-else construction using this code 
Node<T> node;
        for (node = first; node != null; node = node.next) {
            if (object != null && object.equals(node.data) || node.data == object) {
                unlink(node);
                return true;
            }
        }

It looks better
```suggestion
Node<T> node;
        for (node = first; node != null; node = node.next) {
            if (object != null && object.equals(node.data) || node.data == object) {
                unlink(node);
                return true;
            }
        }
```

Reply:
agree, thanks!)